### DISCUSSION TITLE
What is the proper way to fetch related data in a Content Query?

### DISCUSSION THREAD

**Discussion by ssyberg** [statamic/cms#10551] [Q&A]
I'm using Statamic headless so not using antlers and am implementing some custom endpoints. I'm having trouble figuring out the correct way to harness augmentation in my controller. Here's a basic `get` I wrote:

```
    public function get(Request $request)
    {
        $validator = Validator::make($request->all(), [
            'collections' => 'required|string',
        ]);

        if ($validator->fails()) {
            return response()->json(['errors' => $validator->errors()], 400);
        }

        $collections = $request->query('collections');
        $collectionArray = array_filter(explode(',', $collections));
        
        $limit = $request->input('limit', 10);

        $excludeIds = $request->query('exclude_ids', '');
        $excludeIdsArray = array_filter(explode(',', $excludeIds));

        $query = Entry::query()
            ->whereIn('collection', $collectionArray)
            ->limit($limit)
            ->orderByDesc('date');

        if (!empty($excludeIdsArray)) {
            $query->whereNotIn('id', $excludeIdsArray);
        }

        $results = $query->get()->toAugmentedArray();
        
        return response()->json($results);
    }
```

I'm a little blind here as the docs don't really cover any of the details, for example it's not obvious to me what is even returned by `query->get()` but I've tried to call the `shallow` version of Augmentables and get an undefined error. Ideally I'd be able to define what fields and nested fields get included/evaluated in the response on a per collection basis.

P.S. I know what I'm asking for is solvable with GraphQL, but for a number of reasons that's not really an option here.

**Comment by edalzell** [comment]
Like in Laravel, the `get` runs the query.

**Comment by edalzell** [comment]
What's returned from the `get()`?

**Reply by ssyberg** [reply]
The `get` returns a `Statamic\Entries\EntryCollection` according to the error message.

**Reply by edalzell** [reply]
right there you go then, you can use any method that avail on that. You might need to loop through and make the json resource as you like.

**Reply by ssyberg** [reply]
Yea that's what I'm thinking, the meta question here is how do I know what methods are available on an `EntryCollection`? I tried snooping around in the code but there's a lot of indirection and I'm coming up short. The `EntryCollection` does appear to implement `toAugmentedArray` and this is all the places that appears to be defined in the repo:
https://github.com/search?q=repo%3Astatamic%2Fcms+%22function+toAugmentedArray%22&type=code

I had assumed `EntryCollection` was an an [Augmentable](https://github.com/statamic/cms/blob/8efcedcee2f29f36c895e50d4ee48438c4e6f352/src/Contracts/Data/Augmentable.php#L11) which is why I'm stuck became both appear in the Contract but one errors and the other does not. Is it possible this is a bug?
